app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/cloud/BuildDeploymentSourceType.java

```diff
@@ -120,6 +124,22 @@ public void updateBuildBeforeRunOption(
     }
   }
 
+  @Override
+  public void save(@NotNull ModuleDeploymentSource deploymentSource, @NotNull Element tag) {
```

Comment:
Could you please explain logic behind this a bit more - is this saved for deployment - i.e. this is cloud project and service version a user specified?

Reply:
I should have added documentation here.. But essentially save/load are for serializing/deserializing the deployment source.

For example, you create a run config, select a deployment source and set some options, then save it. Then you close the IDE. Later you open the IDE again and want to just run that existing run config. This allows the deployment source info to be restored.

I'll add some javadocs